src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -115,25 +115,40 @@ internal sealed class SourcePropertySymbol : PropertySymbol, IAttributeTargetSym
             bool hasExpressionBody = arrowExpression != null;
             bool hasInitializer = !isIndexer && propertySyntax.Initializer != null;
 
-            bool notRegularProperty = (!IsAbstract && !IsExtern && !isIndexer && hasAccessorList);
+            bool notRegularProperty = !IsAbstract && !IsExtern && !isIndexer && hasAccessorList;
             AccessorDeclarationSyntax getSyntax = null;
             AccessorDeclarationSyntax setSyntax = null;
             if (hasAccessorList)
             {
                 foreach (var accessor in syntax.AccessorList.Accessors)
                 {
-                    if (accessor.Kind() == SyntaxKind.GetAccessorDeclaration &&
-                        (getSyntax == null || getSyntax.Keyword.Span.IsEmpty))
-                    {
-                        getSyntax = accessor;
-                    }
-                    else if (accessor.Kind() == SyntaxKind.SetAccessorDeclaration &&
-                        (setSyntax == null || setSyntax.Keyword.Span.IsEmpty))
-                    {
-                        setSyntax = accessor;
-                    }
-                    else
+                    switch (accessor.Kind())
                     {
+                    case SyntaxKind.GetAccessorDeclaration:
+                        if (getSyntax == null)
+                        {
+                            getSyntax = accessor;
+                        }
+                        else
+                        {
+                            diagnostics.Add(ErrorCode.ERR_DuplicateAccessor, accessor.Keyword.GetLocation());
+                        }
+                        break;
+                    case SyntaxKind.SetAccessorDeclaration:
+                        if (setSyntax == null)
+                        {
+                            setSyntax = accessor;
+                        }
+                        else
+                        {
+                            diagnostics.Add(ErrorCode.ERR_DuplicateAccessor, accessor.Keyword.GetLocation());
+                        }
+                        break;
+                    case SyntaxKind.AddAccessorDeclaration:
+                    case SyntaxKind.RemoveAccessorDeclaration:
+                        diagnostics.Add(ErrorCode.ERR_GetOrSetExpected, accessor.Keyword.GetLocation());
+                        continue;
+                    default:
                         continue;
```

Comment:
Similar comment as in SourceCustomEventSymbol may be useful here (the error was reported by parser).